src/main/java/edu/cornell/kfs/coa/batch/service/impl/CreateClosedAccountsCsvServiceImpl.java

```diff
@@ -0,0 +1,303 @@
+        ensureOutbundDirectoriesExist();
+        
+        LOG.info("writeClosedAccountsToCsvFormattedFile: Prior to obtaining data with SQL query for date range. This may take a while.");
+        List<ClosedAccount> closedAccountsData = getClosedAccountsByDateRangeDao().obtainClosedAccountsDataFor(fileDataRangeSpecified);
+        
+        if (ObjectUtils.isNotNull(closedAccountsData) && closedAccountsData.size() > 0) {
+            LOG.info("writeClosedAccountsToCsvFormattedFile: Have the data. Will now write out " + closedAccountsData.size() + " closed accounts to CSV file.");
+            writeClosedAccountsToCsvFile(closedAccountsData);
+        } else {
+            LOG.info("writeClosedAccountsToCsvFormattedFile: NO CSV file being created. " 
+                    + (ObjectUtils.isNotNull(closedAccountsData) ? (closedAccountsData.size() + " accounts being written to file.") : ("null returned for requested account data.") ));
+        }
+    }
+    
+    private void ensureOutbundDirectoriesExist() {
+        try {
+            FileUtils.forceMkdir(new File(getCsvClosedAccountstFileCreationDirectory()));
+        } catch (IOException e) {
+            LOG.info("ensureOutbundDirectoriesExist: Could not make file creation directory => " + getCsvClosedAccountstFileCreationDirectory() + " Throwing RuntimeException to force batch job failure.");
+            throw new RuntimeException(e);
+        }
+        try {
+            FileUtils.forceMkdir(new File(getCsvClosedAccountsExportDirectory()));
+        } catch (IOException e) {
+            LOG.info("ensureOutbundDirectoriesExist: Could not make file export directory => " + getCsvClosedAccountsExportDirectory() + " Throwing RuntimeException to force batch job failure.");
+            throw new RuntimeException(e);
+        }
+    }
+    
+    private void writeClosedAccountsToCsvFile(List<ClosedAccount> closedAccountsDataList) throws IOException {
+        String csvFileName = generateCsvOutputFileName();
+        String fullyQualifiedCreationDirectoryFileName = fullyQualifyFileNameToCreationDirectory(csvFileName);
+        LOG.info("writeClosedAccountsToCsvFile: fullyQualifiedOutputFile = " + fullyQualifiedCreationDirectoryFileName);
+        File outputFile = new File(fullyQualifiedCreationDirectoryFileName);
+        FileWriter outputFileWriter;
+        outputFileWriter = new FileWriter(outputFile);
+        try {
+            for (ClosedAccount closedAccount : closedAccountsDataList) {
+                outputFileWriter.write(closedAccount.toCsvString());
+                outputFileWriter.write(KFSConstants.NEWLINE);
+            }
+        } catch (IOException io) {
+            LOG.error("writeClosedAccountsToCsvFile: Caught IOException attempting to create CSV file of closded accounts => ", io);
+            throw io;
+        } finally {
+            outputFileWriter.flush();
+            outputFileWriter.close();
```

Comment:
You may want to keep the flush() call at the end of the "try" block for simpler error handling. Also, it would be good to attempt the close() call in a way that will ignore any errors it throws, such as by using the IOUtils.closeQuietly() method.